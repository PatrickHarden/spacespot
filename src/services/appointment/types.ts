export interface AppointmentState {
  data?: Array<Appointment>
  error?: string
}

export interface Appointment {
  id: number
  enquireId: number
  description: string
  peer1Id: string
  peer2Id: string
  proposedTime: string
  createdAt: string
  lastUpdatedAt: string
  lastProposedBy: string
  appointmentStatus: string
}

export interface NewAppointment {
  description: string
  proposedTime: string
}
export interface EditAppointment {
  id: string
  description: string
  proposedTime: string
}
export interface AppointmentAction {
  type: string
  payload?: Array<Appointment> | NewAppointment | EditAppointment | string
}
