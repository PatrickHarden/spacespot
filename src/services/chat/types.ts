export interface ChatAction {
  type: string
  payload?: ChatResponse | ChatMessage | string
}
export interface MessageAction {
  type: string
  payload?: string
}
export interface ChatState {
  data?: ChatResponse
  error?: string
}

export interface ChatResponse {
  chatId: number
  enquiryId: number
  peer1Id: string
  peer2Id: string
  messages: Array<ChatMessage>
}

export interface ChatMessage {
  userId: string
  message: string
  sentAt: string
}
