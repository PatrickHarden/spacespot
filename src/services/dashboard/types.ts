import { Space } from 'services/space/types'

export interface DashboardState {
  buildings: Array<Space>
  spaces: { [key: string]: Array<Space> }
  error?: string
  isLoading: boolean
  initOK: boolean
  pendingDelete: Array<string>
}
