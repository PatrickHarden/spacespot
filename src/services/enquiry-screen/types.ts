export interface EnquiryScreenAction {
  type: string
  payload: EnquiryScreenStateData | string
}
export interface EnquiryScreenState {
  data?: EnquiryScreenStateData
  error?: string
}
export interface EnquiryScreenStateData {
  id: string
}
