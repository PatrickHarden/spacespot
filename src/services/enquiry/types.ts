import { FileItem } from 'services/global/types'
import { NegotiationTerm } from 'services/negotiation/types'

export interface DocumentDTO {
  fileId: number
  fileName: string
  enquireId: number
  uploadAt: Date
  uploadedBy: string
  documentType?: string
}

export const DOC_TYPE_TEMPLATE = 'CompanyTemplate'

export type DocData = any

export interface EnquiryAction {
  type: string
  payload:
    | string
    | number
    | DocumentDTO
    | Array<DocumentDTO>
    | EnquiryResponse
    | FileItem
    | DocData
    | EnquiryRequest
}

export interface EnquiryState {
  data?: EnquiryResponse
  docs?: Array<DocumentDTO>
  error?: string
}

export enum EnquiryType {
  HotDesk = 'HotDesk',
  FixedDesk = 'FixedDesk',
  ServicedOffice = 'ServicedOffice',
  notChecked = '',
}

export interface EnquiryResponse {
  chatId: number
  enquireId: number
  spaceId: string
  /* Tenant */
  peer1Id: string
  /* Landlord */
  peer2Id: string
  negotiationId?: number
  enquireStatus?: string
  flexType?: EnquiryType
  flex: boolean
}

export interface EnquiryAPIResultActions {
  successSignInAfter?: boolean
  errorShowMessage?: string
}

export interface EnquiryRequest {
  spaceId: string
  message: string
  negotiation: {
    flexible: boolean
    terms: Array<NegotiationTerm>
  }
  userId?: string
  loggedIn?: boolean
  apiResultActions?: EnquiryAPIResultActions
}

export interface EnquiryRequestFlex {
  spaceId: string
  message: string
  flex: boolean
  flexType: EnquiryType
  flexIdentifier?: string
  negotiation: {
    terms: Array<NegotiationTerm>
  }
  userId?: string
  loggedIn?: boolean
  apiResultActions?: EnquiryAPIResultActions
}
