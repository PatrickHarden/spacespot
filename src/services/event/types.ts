export interface EnquiryEvent {
  eventId: number
  eventMessage: any
  eventType: string
  eventDate: Date
}

export interface EnquiryEventState {
  [enquiryId: string]: {
    events: Array<EnquiryEvent>
    isLoading?: boolean
    error?: string
  }
}

export interface GetEnquiryEventsAction {
  type: string
  payload: {
    enquiryId: string
  }
}

export interface GetEnquiryEventsSuccessAction {
  type: string
  payload: {
    enquiryId: string
    events: Array<EnquiryEvent>
  }
}

export interface GetEnquiryEventsErrorAction {
  type: string
  payload: {
    enquiryId: string
    error: string
  }
}

export type EnquiryEventAction =
  | GetEnquiryEventsAction
  | GetEnquiryEventsSuccessAction
  | GetEnquiryEventsErrorAction
