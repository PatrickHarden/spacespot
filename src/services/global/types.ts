import { ChatState } from 'services/chat/types'
import { EnquiryState } from 'services/enquiry/types'
import { EnquiryScreenState } from 'services/enquiry-screen/types'
import { SpaceState } from 'services/space/types'
import { UserState } from 'services/user/types'
import { AppointmentState } from 'services/appointment/types'
import { OnBoardingState } from 'services/onboarding/types'
import { NegotiationState } from 'services/negotiation/types'
import { DashboardState } from 'services/dashboard/types'
import { MyEnquiriesState } from 'services/myEnquiries/types'
import { SearchState } from 'services/search/types'

export interface SpacespotState {
  chat?: ChatState
  dashboard?: DashboardState
  enquiry?: EnquiryState
  enquiryscreen?: EnquiryScreenState
  space?: SpaceState
  user?: UserState
  appointment?: AppointmentState
  onboarding?: OnBoardingState
  negotiation?: NegotiationState
  myenquiries?: MyEnquiriesState
  search?: SearchState
}

export interface SpacespotErrorApi {
  error?: any
  message?: any
}

export interface FileItem {
  key: number
  uri: string
  name: string
  type: string
}
