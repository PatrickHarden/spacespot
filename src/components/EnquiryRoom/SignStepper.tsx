import React from 'react'
import { useIntl } from 'react-intl'
import styled from 'styled-components'

import Stepper from '@material-ui/core/Stepper'
import Step from '@material-ui/core/Step'
import StepLabel from '@material-ui/core/StepLabel'
import { NegotiationStatus } from 'services/negotiation/types'

const CustomStep = styled(Step)`
  .MuiStepIcon-text {
    font-size: 14px;
    font-weight: 400;
  }
  .MuiStepIcon-root {
    color: white;
    border: 1px solid black;
    border-radius: 50%;
  }
  .MuiStepIcon-completed {
    color: #404042;
    border: none;
  }
  .MuiStepIcon-active {
    border: none;
    color: #404042;
    .MuiStepIcon-text {
      fill: #ffffff;
    }
  }
  .MuiStepLabel-label.MuiStepLabel-alternativeLabel {
    margin-top: 2px;
  }
  .MuiStepLabel-label.MuiStepLabel-alternativeLabel.MuiTypography-body2 {
    font-size: 14px;
  }
`

const CustomStep2 = styled(CustomStep)`
  .MuiStepIcon-active {
    border: 1px solid black;
    color: #ffffff;
    .MuiStepIcon-text {
      fill: #000000;
    }
  }
`

const SignStepper = styled(
  (props: {
    className?: string
    userStatus: NegotiationStatus
    peerStatus: NegotiationStatus
  }) => {
    const { formatMessage } = useIntl()
    const t = (s: string) => formatMessage({ id: s })
    const { className, userStatus, peerStatus } = props
    let currentInWhite = false
    let signStep = 0
    if (userStatus === NegotiationStatus.SignInfo) {
      signStep = 0
    }
    if (userStatus === NegotiationStatus.AcceptLease) {
      signStep = 1
    }
    if (
      userStatus === NegotiationStatus.LeaseAccepted &&
      peerStatus !== NegotiationStatus.LeaseAccepted
    ) {
      signStep = 2
      currentInWhite = true
    }
    if (
      userStatus === NegotiationStatus.LeaseAccepted &&
      peerStatus === NegotiationStatus.LeaseAccepted
    ) {
      signStep = 2
    }
    if (
      userStatus === NegotiationStatus.LeaseSigned ||
      userStatus === NegotiationStatus.LeaseSignInitiated
    ) {
      signStep = 2
    }
    return (
      <Stepper className={className} activeStep={signStep} alternativeLabel>
        <CustomStep key={1}>
          <StepLabel>{t('SUMMARY_SIGNING_DETAILS')}</StepLabel>
        </CustomStep>
        <CustomStep key={2}>
          <StepLabel>{t('SUMMARY_SIGNING_ACCEPT')}</StepLabel>
        </CustomStep>
        {currentInWhite ? (
          <CustomStep2 key={3}>
            <StepLabel>{t('SUMMARY_SIGNING_SIGN')}</StepLabel>
          </CustomStep2>
        ) : (
          <CustomStep key={3}>
            <StepLabel>{t('SUMMARY_SIGNING_SIGN')}</StepLabel>
          </CustomStep>
        )}
      </Stepper>
    )
  },
)`
  padding: 20px 0;
  background: #fafafa;
`

export default SignStepper
