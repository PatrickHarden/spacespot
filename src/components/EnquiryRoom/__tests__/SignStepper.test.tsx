import React from 'react'
import { render } from '@testing-library/react'

import SignStepper from '../SignStepper'

import { NegotiationStatus } from 'services/negotiation/types'

jest.mock('react-intl', () => ({
  useIntl: () => ({ formatMessage: (s: any) => s.id }),
  createIntlCache: () => jest.fn(),
  createIntl: () => ({ formatMessage: (s: any) => s.id }),
}))

const mockDispatch = jest.fn()

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: () => mockDispatch,
}))

it('Renders without crash', () => {
  const wrapper = render(
    <SignStepper
      userStatus={NegotiationStatus.LeaseAccepted}
      peerStatus={NegotiationStatus.LeaseAccepted}
    />,
  )
  expect(wrapper.asFragment()).toMatchSnapshot()
})

it('Case SignInfo', () => {
  const wrapper = render(
    <SignStepper
      userStatus={NegotiationStatus.SignInfo}
      peerStatus={NegotiationStatus.LeaseAccepted}
    />,
  )
  expect(wrapper.asFragment()).toMatchSnapshot()
})

it('Case AcceptLease', () => {
  const wrapper = render(
    <SignStepper
      userStatus={NegotiationStatus.AcceptLease}
      peerStatus={NegotiationStatus.LeaseAccepted}
    />,
  )
  expect(wrapper.asFragment()).toMatchSnapshot()
})

it('Case peer not LeaseAccepted', () => {
  const wrapper = render(
    <SignStepper
      userStatus={NegotiationStatus.LeaseAccepted}
      peerStatus={NegotiationStatus.SignInfo}
    />,
  )
  expect(wrapper.asFragment()).toMatchSnapshot()
})

it('Case peer LeaseSignInitiated', () => {
  const wrapper = render(
    <SignStepper
      userStatus={NegotiationStatus.LeaseSigned}
      peerStatus={NegotiationStatus.LeaseSignInitiated}
    />,
  )
  expect(wrapper.asFragment()).toMatchSnapshot()
})
