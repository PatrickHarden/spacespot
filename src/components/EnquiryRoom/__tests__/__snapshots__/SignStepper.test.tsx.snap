// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Case AcceptLease 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation0 MuiStepper-root MuiStepper-horizontal sc-htpNat hbskFm MuiStepper-alternativeLabel"
  >
    <div
      class="MuiStep-root MuiStep-horizontal sc-bdVaJa DWZdo MuiStep-alternativeLabel MuiStep-completed"
    >
      <span
        class="MuiStepLabel-root MuiStepLabel-horizontal MuiStepLabel-alternativeLabel"
      >
        <span
          class="MuiStepLabel-iconContainer MuiStepLabel-alternativeLabel"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiStepIcon-root MuiStepIcon-completed"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 0a12 12 0 1 0 0 24 12 12 0 0 0 0-24zm-2 17l-5-5 1.4-1.4 3.6 3.6 7.6-7.6L19 8l-9 9z"
            />
          </svg>
        </span>
        <span
          class="MuiStepLabel-labelContainer"
        >
          <span
            class="MuiTypography-root MuiStepLabel-label MuiStepLabel-alternativeLabel MuiStepLabel-completed MuiTypography-body2 MuiTypography-displayBlock"
          >
            SUMMARY_SIGNING_DETAILS
          </span>
        </span>
      </span>
    </div>
    <div
      class="MuiStep-root MuiStep-horizontal sc-bdVaJa DWZdo MuiStep-alternativeLabel"
    >
      <div
        class="MuiStepConnector-root MuiStepConnector-horizontal MuiStepConnector-alternativeLabel MuiStepConnector-active"
      >
        <span
          class="MuiStepConnector-line MuiStepConnector-lineHorizontal"
        />
      </div>
      <span
        class="MuiStepLabel-root MuiStepLabel-horizontal MuiStepLabel-alternativeLabel"
      >
        <span
          class="MuiStepLabel-iconContainer MuiStepLabel-alternativeLabel"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiStepIcon-root MuiStepIcon-active"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <circle
              cx="12"
              cy="12"
              r="12"
            />
            <text
              class="MuiStepIcon-text"
              text-anchor="middle"
              x="12"
              y="16"
            >
              2
            </text>
          </svg>
        </span>
        <span
          class="MuiStepLabel-labelContainer"
        >
          <span
            class="MuiTypography-root MuiStepLabel-label MuiStepLabel-alternativeLabel MuiStepLabel-active MuiTypography-body2 MuiTypography-displayBlock"
          >
            SUMMARY_SIGNING_ACCEPT
          </span>
        </span>
      </span>
    </div>
    <div
      class="MuiStep-root MuiStep-horizontal sc-bdVaJa DWZdo MuiStep-alternativeLabel"
    >
      <div
        class="MuiStepConnector-root MuiStepConnector-horizontal MuiStepConnector-alternativeLabel Mui-disabled"
      >
        <span
          class="MuiStepConnector-line MuiStepConnector-lineHorizontal"
        />
      </div>
      <span
        class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled MuiStepLabel-alternativeLabel"
      >
        <span
          class="MuiStepLabel-iconContainer MuiStepLabel-alternativeLabel"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiStepIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <circle
              cx="12"
              cy="12"
              r="12"
            />
            <text
              class="MuiStepIcon-text"
              text-anchor="middle"
              x="12"
              y="16"
            >
              3
            </text>
          </svg>
        </span>
        <span
          class="MuiStepLabel-labelContainer"
        >
          <span
            class="MuiTypography-root MuiStepLabel-label MuiStepLabel-alternativeLabel MuiTypography-body2 MuiTypography-displayBlock"
          >
            SUMMARY_SIGNING_SIGN
          </span>
        </span>
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Case SignInfo 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation0 MuiStepper-root MuiStepper-horizontal sc-htpNat hbskFm MuiStepper-alternativeLabel"
  >
    <div
      class="MuiStep-root MuiStep-horizontal sc-bdVaJa DWZdo MuiStep-alternativeLabel"
    >
      <span
        class="MuiStepLabel-root MuiStepLabel-horizontal MuiStepLabel-alternativeLabel"
      >
        <span
          class="MuiStepLabel-iconContainer MuiStepLabel-alternativeLabel"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiStepIcon-root MuiStepIcon-active"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <circle
              cx="12"
              cy="12"
              r="12"
            />
            <text
              class="MuiStepIcon-text"
              text-anchor="middle"
              x="12"
              y="16"
            >
              1
            </text>
          </svg>
        </span>
        <span
          class="MuiStepLabel-labelContainer"
        >
          <span
            class="MuiTypography-root MuiStepLabel-label MuiStepLabel-alternativeLabel MuiStepLabel-active MuiTypography-body2 MuiTypography-displayBlock"
          >
            SUMMARY_SIGNING_DETAILS
          </span>
        </span>
      </span>
    </div>
    <div
      class="MuiStep-root MuiStep-horizontal sc-bdVaJa DWZdo MuiStep-alternativeLabel"
    >
      <div
        class="MuiStepConnector-root MuiStepConnector-horizontal MuiStepConnector-alternativeLabel Mui-disabled"
      >
        <span
          class="MuiStepConnector-line MuiStepConnector-lineHorizontal"
        />
      </div>
      <span
        class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled MuiStepLabel-alternativeLabel"
      >
        <span
          class="MuiStepLabel-iconContainer MuiStepLabel-alternativeLabel"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiStepIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <circle
              cx="12"
              cy="12"
              r="12"
            />
            <text
              class="MuiStepIcon-text"
              text-anchor="middle"
              x="12"
              y="16"
            >
              2
            </text>
          </svg>
        </span>
        <span
          class="MuiStepLabel-labelContainer"
        >
          <span
            class="MuiTypography-root MuiStepLabel-label MuiStepLabel-alternativeLabel MuiTypography-body2 MuiTypography-displayBlock"
          >
            SUMMARY_SIGNING_ACCEPT
          </span>
        </span>
      </span>
    </div>
    <div
      class="MuiStep-root MuiStep-horizontal sc-bdVaJa DWZdo MuiStep-alternativeLabel"
    >
      <div
        class="MuiStepConnector-root MuiStepConnector-horizontal MuiStepConnector-alternativeLabel Mui-disabled"
      >
        <span
          class="MuiStepConnector-line MuiStepConnector-lineHorizontal"
        />
      </div>
      <span
        class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled MuiStepLabel-alternativeLabel"
      >
        <span
          class="MuiStepLabel-iconContainer MuiStepLabel-alternativeLabel"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiStepIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <circle
              cx="12"
              cy="12"
              r="12"
            />
            <text
              class="MuiStepIcon-text"
              text-anchor="middle"
              x="12"
              y="16"
            >
              3
            </text>
          </svg>
        </span>
        <span
          class="MuiStepLabel-labelContainer"
        >
          <span
            class="MuiTypography-root MuiStepLabel-label MuiStepLabel-alternativeLabel MuiTypography-body2 MuiTypography-displayBlock"
          >
            SUMMARY_SIGNING_SIGN
          </span>
        </span>
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Case peer LeaseSignInitiated 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation0 MuiStepper-root MuiStepper-horizontal sc-htpNat hbskFm MuiStepper-alternativeLabel"
  >
    <div
      class="MuiStep-root MuiStep-horizontal sc-bdVaJa DWZdo MuiStep-alternativeLabel MuiStep-completed"
    >
      <span
        class="MuiStepLabel-root MuiStepLabel-horizontal MuiStepLabel-alternativeLabel"
      >
        <span
          class="MuiStepLabel-iconContainer MuiStepLabel-alternativeLabel"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiStepIcon-root MuiStepIcon-completed"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 0a12 12 0 1 0 0 24 12 12 0 0 0 0-24zm-2 17l-5-5 1.4-1.4 3.6 3.6 7.6-7.6L19 8l-9 9z"
            />
          </svg>
        </span>
        <span
          class="MuiStepLabel-labelContainer"
        >
          <span
            class="MuiTypography-root MuiStepLabel-label MuiStepLabel-alternativeLabel MuiStepLabel-completed MuiTypography-body2 MuiTypography-displayBlock"
          >
            SUMMARY_SIGNING_DETAILS
          </span>
        </span>
      </span>
    </div>
    <div
      class="MuiStep-root MuiStep-horizontal sc-bdVaJa DWZdo MuiStep-alternativeLabel MuiStep-completed"
    >
      <div
        class="MuiStepConnector-root MuiStepConnector-horizontal MuiStepConnector-alternativeLabel MuiStepConnector-completed"
      >
        <span
          class="MuiStepConnector-line MuiStepConnector-lineHorizontal"
        />
      </div>
      <span
        class="MuiStepLabel-root MuiStepLabel-horizontal MuiStepLabel-alternativeLabel"
      >
        <span
          class="MuiStepLabel-iconContainer MuiStepLabel-alternativeLabel"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiStepIcon-root MuiStepIcon-completed"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 0a12 12 0 1 0 0 24 12 12 0 0 0 0-24zm-2 17l-5-5 1.4-1.4 3.6 3.6 7.6-7.6L19 8l-9 9z"
            />
          </svg>
        </span>
        <span
          class="MuiStepLabel-labelContainer"
        >
          <span
            class="MuiTypography-root MuiStepLabel-label MuiStepLabel-alternativeLabel MuiStepLabel-completed MuiTypography-body2 MuiTypography-displayBlock"
          >
            SUMMARY_SIGNING_ACCEPT
          </span>
        </span>
      </span>
    </div>
    <div
      class="MuiStep-root MuiStep-horizontal sc-bdVaJa DWZdo MuiStep-alternativeLabel"
    >
      <div
        class="MuiStepConnector-root MuiStepConnector-horizontal MuiStepConnector-alternativeLabel MuiStepConnector-active"
      >
        <span
          class="MuiStepConnector-line MuiStepConnector-lineHorizontal"
        />
      </div>
      <span
        class="MuiStepLabel-root MuiStepLabel-horizontal MuiStepLabel-alternativeLabel"
      >
        <span
          class="MuiStepLabel-iconContainer MuiStepLabel-alternativeLabel"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiStepIcon-root MuiStepIcon-active"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <circle
              cx="12"
              cy="12"
              r="12"
            />
            <text
              class="MuiStepIcon-text"
              text-anchor="middle"
              x="12"
              y="16"
            >
              3
            </text>
          </svg>
        </span>
        <span
          class="MuiStepLabel-labelContainer"
        >
          <span
            class="MuiTypography-root MuiStepLabel-label MuiStepLabel-alternativeLabel MuiStepLabel-active MuiTypography-body2 MuiTypography-displayBlock"
          >
            SUMMARY_SIGNING_SIGN
          </span>
        </span>
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Case peer not LeaseAccepted 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation0 MuiStepper-root MuiStepper-horizontal sc-htpNat hbskFm MuiStepper-alternativeLabel"
  >
    <div
      class="MuiStep-root MuiStep-horizontal sc-bdVaJa DWZdo MuiStep-alternativeLabel MuiStep-completed"
    >
      <span
        class="MuiStepLabel-root MuiStepLabel-horizontal MuiStepLabel-alternativeLabel"
      >
        <span
          class="MuiStepLabel-iconContainer MuiStepLabel-alternativeLabel"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiStepIcon-root MuiStepIcon-completed"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 0a12 12 0 1 0 0 24 12 12 0 0 0 0-24zm-2 17l-5-5 1.4-1.4 3.6 3.6 7.6-7.6L19 8l-9 9z"
            />
          </svg>
        </span>
        <span
          class="MuiStepLabel-labelContainer"
        >
          <span
            class="MuiTypography-root MuiStepLabel-label MuiStepLabel-alternativeLabel MuiStepLabel-completed MuiTypography-body2 MuiTypography-displayBlock"
          >
            SUMMARY_SIGNING_DETAILS
          </span>
        </span>
      </span>
    </div>
    <div
      class="MuiStep-root MuiStep-horizontal sc-bdVaJa DWZdo MuiStep-alternativeLabel MuiStep-completed"
    >
      <div
        class="MuiStepConnector-root MuiStepConnector-horizontal MuiStepConnector-alternativeLabel MuiStepConnector-completed"
      >
        <span
          class="MuiStepConnector-line MuiStepConnector-lineHorizontal"
        />
      </div>
      <span
        class="MuiStepLabel-root MuiStepLabel-horizontal MuiStepLabel-alternativeLabel"
      >
        <span
          class="MuiStepLabel-iconContainer MuiStepLabel-alternativeLabel"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiStepIcon-root MuiStepIcon-completed"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 0a12 12 0 1 0 0 24 12 12 0 0 0 0-24zm-2 17l-5-5 1.4-1.4 3.6 3.6 7.6-7.6L19 8l-9 9z"
            />
          </svg>
        </span>
        <span
          class="MuiStepLabel-labelContainer"
        >
          <span
            class="MuiTypography-root MuiStepLabel-label MuiStepLabel-alternativeLabel MuiStepLabel-completed MuiTypography-body2 MuiTypography-displayBlock"
          >
            SUMMARY_SIGNING_ACCEPT
          </span>
        </span>
      </span>
    </div>
    <div
      class="MuiStep-root MuiStep-horizontal sc-bdVaJa sc-bwzfXH jYVzVW MuiStep-alternativeLabel"
    >
      <div
        class="MuiStepConnector-root MuiStepConnector-horizontal MuiStepConnector-alternativeLabel MuiStepConnector-active"
      >
        <span
          class="MuiStepConnector-line MuiStepConnector-lineHorizontal"
        />
      </div>
      <span
        class="MuiStepLabel-root MuiStepLabel-horizontal MuiStepLabel-alternativeLabel"
      >
        <span
          class="MuiStepLabel-iconContainer MuiStepLabel-alternativeLabel"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiStepIcon-root MuiStepIcon-active"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <circle
              cx="12"
              cy="12"
              r="12"
            />
            <text
              class="MuiStepIcon-text"
              text-anchor="middle"
              x="12"
              y="16"
            >
              3
            </text>
          </svg>
        </span>
        <span
          class="MuiStepLabel-labelContainer"
        >
          <span
            class="MuiTypography-root MuiStepLabel-label MuiStepLabel-alternativeLabel MuiStepLabel-active MuiTypography-body2 MuiTypography-displayBlock"
          >
            SUMMARY_SIGNING_SIGN
          </span>
        </span>
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Renders without crash 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation0 MuiStepper-root MuiStepper-horizontal sc-htpNat hbskFm MuiStepper-alternativeLabel"
  >
    <div
      class="MuiStep-root MuiStep-horizontal sc-bdVaJa DWZdo MuiStep-alternativeLabel MuiStep-completed"
    >
      <span
        class="MuiStepLabel-root MuiStepLabel-horizontal MuiStepLabel-alternativeLabel"
      >
        <span
          class="MuiStepLabel-iconContainer MuiStepLabel-alternativeLabel"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiStepIcon-root MuiStepIcon-completed"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 0a12 12 0 1 0 0 24 12 12 0 0 0 0-24zm-2 17l-5-5 1.4-1.4 3.6 3.6 7.6-7.6L19 8l-9 9z"
            />
          </svg>
        </span>
        <span
          class="MuiStepLabel-labelContainer"
        >
          <span
            class="MuiTypography-root MuiStepLabel-label MuiStepLabel-alternativeLabel MuiStepLabel-completed MuiTypography-body2 MuiTypography-displayBlock"
          >
            SUMMARY_SIGNING_DETAILS
          </span>
        </span>
      </span>
    </div>
    <div
      class="MuiStep-root MuiStep-horizontal sc-bdVaJa DWZdo MuiStep-alternativeLabel MuiStep-completed"
    >
      <div
        class="MuiStepConnector-root MuiStepConnector-horizontal MuiStepConnector-alternativeLabel MuiStepConnector-completed"
      >
        <span
          class="MuiStepConnector-line MuiStepConnector-lineHorizontal"
        />
      </div>
      <span
        class="MuiStepLabel-root MuiStepLabel-horizontal MuiStepLabel-alternativeLabel"
      >
        <span
          class="MuiStepLabel-iconContainer MuiStepLabel-alternativeLabel"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiStepIcon-root MuiStepIcon-completed"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 0a12 12 0 1 0 0 24 12 12 0 0 0 0-24zm-2 17l-5-5 1.4-1.4 3.6 3.6 7.6-7.6L19 8l-9 9z"
            />
          </svg>
        </span>
        <span
          class="MuiStepLabel-labelContainer"
        >
          <span
            class="MuiTypography-root MuiStepLabel-label MuiStepLabel-alternativeLabel MuiStepLabel-completed MuiTypography-body2 MuiTypography-displayBlock"
          >
            SUMMARY_SIGNING_ACCEPT
          </span>
        </span>
      </span>
    </div>
    <div
      class="MuiStep-root MuiStep-horizontal sc-bdVaJa DWZdo MuiStep-alternativeLabel"
    >
      <div
        class="MuiStepConnector-root MuiStepConnector-horizontal MuiStepConnector-alternativeLabel MuiStepConnector-active"
      >
        <span
          class="MuiStepConnector-line MuiStepConnector-lineHorizontal"
        />
      </div>
      <span
        class="MuiStepLabel-root MuiStepLabel-horizontal MuiStepLabel-alternativeLabel"
      >
        <span
          class="MuiStepLabel-iconContainer MuiStepLabel-alternativeLabel"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiStepIcon-root MuiStepIcon-active"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <circle
              cx="12"
              cy="12"
              r="12"
            />
            <text
              class="MuiStepIcon-text"
              text-anchor="middle"
              x="12"
              y="16"
            >
              3
            </text>
          </svg>
        </span>
        <span
          class="MuiStepLabel-labelContainer"
        >
          <span
            class="MuiTypography-root MuiStepLabel-label MuiStepLabel-alternativeLabel MuiStepLabel-active MuiTypography-body2 MuiTypography-displayBlock"
          >
            SUMMARY_SIGNING_SIGN
          </span>
        </span>
      </span>
    </div>
  </div>
</DocumentFragment>
`;
